exes/mqtt-broadcaster/inc/mqtt_broadcaster.hpp

```diff
@@ -0,0 +1,560 @@
+  }
+
+  auto set_sni_hostname(auto&& native_handle, std::string broker_address) -> void {
+    if (!SSL_set_tlsext_host_name(native_handle, broker_address.c_str())) {
+      boost::system::error_code error{ static_cast<int>(::ERR_get_error()), boost::asio::error::get_ssl_category() };
+
+      throw boost::system::system_error{ error };
+    }
+  }
+
+  auto async_handshake(auto&& socket) -> asio::awaitable<void> {
+    co_await socket.async_handshake(async_mqtt::tls::stream_base::client, asio::use_awaitable);
+  }
+};
+
+struct signal_data {
+  tfc::ipc_ruler::signal information;
+  tfc::ipc::details::any_recv receiver;
+  std::optional<std::any> current_value;
+};
+
+template <class ipc_client_type, class mqtt_client_type, class config_type, class network_manager_type>
+class mqtt_broadcaster {
+public:
+  static constexpr std::string_view namespace_element = "spBv1.0";
+  static constexpr std::string_view ndata = "NDATA";
+  static constexpr std::string_view nbirth = "NBIRTH";
+  static constexpr std::string_view ndeath = "NDEATH";
+  static constexpr std::string_view ncmd = "NCMD";
+  static constexpr std::string_view rebirth_metric = "Node Control/Rebirth";
+
+  explicit mqtt_broadcaster(asio::io_context& io_ctx)
+      : io_ctx_(io_ctx), tls_ctx_{ async_mqtt::tls::context::tlsv12 }, ipc_client_{ io_ctx_ },
+        mqtt_client_(std::make_shared<mqtt_client_type>(async_mqtt::protocol_version::v5, io_ctx.get_executor(), tls_ctx_)),
+        logger_("mqtt_broadcaster"), config_(io_ctx, "mqtt_broadcaster"), network_manager_() {
+    properties_callback_ = ipc_client_.register_properties_change_callback(
+        [this]([[maybe_unused]] sdbusplus::message_t& msg) { add_new_signals(); });
+    asio::co_spawn(mqtt_client_->strand(), initialize(), asio::detached);
+  }
+
+  explicit mqtt_broadcaster(asio::io_context& io_ctx, bool)
+      : io_ctx_(io_ctx), tls_ctx_{ async_mqtt::tls::context::tlsv12 }, ipc_client_{ io_ctx_ },
+        mqtt_client_(std::make_shared<mqtt_client_type>(async_mqtt::protocol_version::v5, io_ctx.get_executor(), tls_ctx_)),
+        logger_("mqtt_broadcaster"), config_(io_ctx, "mqtt_broadcaster"), network_manager_() {}
+
+private:
+  auto initialize() -> asio::awaitable<void> {
+    co_await asio::co_spawn(mqtt_client_->strand(), connect_to_broker(), asio::use_awaitable);
+    add_new_signals();
+    asio::co_spawn(mqtt_client_->strand(), ncmd_listener(), asio::detached);
+  }
+
+  auto timestamp_milliseconds() -> std::chrono::milliseconds {
+    using std::chrono::duration_cast;
+    using std::chrono::milliseconds;
+    using std::chrono::system_clock;
+    return duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+  }
+
+  auto get_payload() -> Payload {
+    Payload payload;
+    payload.set_timestamp(timestamp_milliseconds().count());
+    if (seq_ == 255) {
+      seq_ = 0;
+    } else {
+      seq_++;
+    }
+
+    payload.set_seq(seq_);
```

Comment:
maybe rename function to make_payload as the operation `get` is antonym of `set` here